projects/smoke_ads1259/src/main.c

```diff
@@ -1,15 +1,23 @@
+#include <stdlib.h>
+
 #include "ads1259_adc.h"
 #include "delay.h"
 #include "gpio.h"
 #include "interrupt.h"
 #include "log.h"
 #include "soft_timer.h"
 
+// change this number to read for x cycles, 0 for infinite
+#define READ_CYCLE_NUM 2
+// change this number to read x times in a cycle
+#define READ_CYCLE_SIZE 3
+
 // slightly larger than conversion time of adc
 #define CONVERSION_TIME_MS 18
-#define READING_QUEUE_LENGTH 10
+#define READING_QUEUE_LENGTH READ_CYCLE_SIZE
 static Ads1259Storage s_storage;
 static int s_index;
+static int s_count = 1;
```

Comment:
Minor nit: these should probably be `uint8_t`s.

Reply:
changed to `uint8_t`